mockfirestore/document.py

```diff
@@ -70,3 +73,41 @@ def collection(self, name) -> 'CollectionReference':
         if name not in document:
             set_by_path(self._data, new_path, {})
         return CollectionReference(self._data, new_path, parent=self)
+
+
+def _apply_transformations(document: Dict[str, Any], data: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
```suggestion
def _apply_transformations(document: Dict[str, Any], data: Dict[str, Any]):
```

This function doesn't actually return anything.

Reply:
Also it would be good to have it be a private method inside `DocumentSnapshot`, to be consistent with the other classes.

Reply:
I explicitly kept this out of the class because it seemed pretty strange to have this helper functionality available to instances of `DocumentReference`. This way, it's more "private" than a private method in the class definition, since it doesn't follow the instance everywhere it is used.

e.g. doesn't allow for this

```
from mockfirestore import DocumentReference
DocumentReference(...)._apply_transformations(...)
```

Furthermore, the way that it's currently written, it doesn't explicitly rely on the `DocumentReference` -- only the `document` object itself. Thus, I opted for composition over (potential) inheritance in this case.